### ISSUE TITLE
generate batch data from 1 big image

### ISSUE THREAD

**Issue by Nic-Ma** [Project-MONAI/MONAI#130]
**Is your feature request related to a problem? Please describe.**
As many medical images are big and hard to label(especially segmentation), most of the medical training datasets contain only 10 - 100 big images. We need to crop 1 big image to generate a batch of smaller images for training.

**Describe the solution you'd like**
There are many challenges in this task:
1. After cropping, the output data shape will be [N, C, H, W, D] instead of original [C, H, W, D].
    And still need to apply other transforms to every output data.
2. In the DataLoader, got a batch of data in [N, C, H, W, D], need to reshape to [batch x N, C, H, W, D].
3. Need to support many different methods to crop expected small batch:
    (1) Use the sliding window to crop image & label together.
    (2) Randomly crop image & label based on positive/negative ratio(use label field).
    ... ...

**Describe alternatives you've considered**
In TorchVsion's vanilla transforms, there is a very basic reference method:
```py
transform = Compose([
    TenCrop(size), # this is a list of PIL Images
    # returns a 4D tensor
    Lambda(lambda crops: torch.stack([ToTensor()(crop) for crop in crops]))
])
```
